app/src/androidTest/java/org/hisp/dhis/android/dataentry/main/home/HomeRepositoryIntegrationTests.java

```diff
@@ -54,16 +54,19 @@ public void setUp() throws IOException {
     }
 
     @Test
-    public void homeEntitiesShouldReturnCorrectTypes() {
+    public void homeViewModelsShouldReturnCorrectTypes() {
 
         databaseRule.database().insert(TrackedEntityModel.TABLE, null, trackedEntity);
         databaseRule.database().insert(ProgramModel.TABLE, null, program);
 
-        TestObserver<List<HomeViewModel>> testObserver = homeRepository.homeEntities().test();
+        // this program should not be returned
+        databaseRule.database().insert(ProgramModel.TABLE, null, programWithRegistration);
+
+        TestObserver<List<HomeViewModel>> testObserver = homeRepository.homeViewModels().test();
 
         testObserver.assertValueCount(1); // One list is returned
```

Comment:
Okay, after viewing the file I found those assertions at the end. Maybe it could be better to group them?